src/main/webapp/app/oncokb-annotation-visualisation/AnnotationVisualisation.tsx

```diff
@@ -0,0 +1,380 @@
+    );
+  }
+
+  @computed
+  get copyNumberTreatments(): TreatmentImplication[] {
+    return this.props.treatments.filter(
+      treatment => treatment.alterationType === 'COPY_NUMBER_ALTERATION'
+    );
+  }
+
+  @computed
+  get structuralTreatments(): TreatmentImplication[] {
+    return this.props.treatments.filter(
+      treatment => treatment.alterationType === 'STRUCTURAL_VARIANT'
+    );
+  }
+
+  @computed
+  get mutationsNotifications(): NotificationImplication[] {
+    return this.props.notifications.filter(
+      notification => notification.alterationType === 'MUTATION'
+    );
+  }
+
+  @computed
+  get copyNumberNotifications(): NotificationImplication[] {
+    return this.props.notifications.filter(
+      notification => notification.alterationType === 'COPY_NUMBER_ALTERATION'
+    );
+  }
+
+  @computed
+  get structuralNotifications(): NotificationImplication[] {
+    return this.props.notifications.filter(
+      notification => notification.alterationType === 'STRUCTURAL_VARIANT'
+    );
+  }
+
+  constructor(props: AnnotationVisualisationProps) {
+    super(props);
+    this.state = {
+      selectedAnnotationColumns: defaultAnnotationColumns,
+      selectedTreatmentColumns: defaultTreatmentColumns,
+    };
+  }
+
+  componentDidMount() {
+    const savedAnnotationColumns = localStorage.getItem(
+      'selectedAnnotationColumns'
+    );
+    const savedTreatmentColumns = localStorage.getItem(
+      'selectedTreatmentColumns'
+    );
+
+    if (savedAnnotationColumns && savedTreatmentColumns) {
+      this.setState({
+        selectedAnnotationColumns: JSON.parse(savedAnnotationColumns),
+        selectedTreatmentColumns: JSON.parse(savedTreatmentColumns),
+      });
+    }
+  }
+
+  handleAnnotationColumnsChange = (selectedOptions: any) => {
+    this.setState({
+      selectedAnnotationColumns: selectedOptions.map((option: any) =>
+        option.value ? option.value : option.prop
+      ),
+    });
+    localStorage.setItem(
+      'selectedAnnotationColumns',
+      JSON.stringify(selectedOptions.map((option: any) => option.value))
+    );
+  };
+
+  handleTreatmentColumnsChange = (selectedOptions: any) => {
+    this.setState({
+      selectedTreatmentColumns: selectedOptions.map(
+        (option: any) => option.value
+      ),
+    });
+    localStorage.setItem(
+      'selectedTreatmentColumns',
+      JSON.stringify(selectedOptions.map((option: any) => option.value))
+    );
+  };
+
+  @computed
+  get treatmentTableColumns(): SearchColumn<TreatmentImplication>[] {
+    const selectedKeys = this.state.selectedTreatmentColumns;
+    const filteredColumns = treatmentColumns.filter(col =>
+      selectedKeys.includes(col.key)
+    );
+    const hasAnnotation = selectedKeys.some(
+      column => column === TREATMENTS_TABLE_COLUMN_KEY.ANNOTATION
+    );
+    const totalColumns = selectedKeys.length;
+    const calculatedWidth = hasAnnotation
+      ? 800 / (totalColumns - 1)
```

Comment:
Again, I'd be careful with these hardcoded widths. See my other comment for details.